server/cronJobs/aggregateJob.js

```diff
@@ -0,0 +1,58 @@
+import { getAllCategories } from '@server/daos/products';
+import {
+  getCountByDate,
+  getCountByDateForCategory,
+  getEarliestCreatedDate,
+  getTotalByDate,
+  getTotalByDateForCategory
+} from '@server/daos/purchasedProducts';
+import { redis } from '@server/services/redis';
+import { logger } from '@server/utils';
+import { REDIS_IMPLEMENTATION_DATE } from '@server/utils/constants';
+import moment from 'moment';
+
+export const aggregateCheck = async () => {
+  let startDate;
+  const previousDate = moment()
+    .subtract(1, 'day')
+    .format('YYYY-MM-DD');
+  const endDate = REDIS_IMPLEMENTATION_DATE || previousDate;
+  const lastSyncFor = await redis.get('lastSyncFor');
```

Comment:
lets check for conditions where it shouldn't run and then we'll `return;` from there